### PR TITLE
Update dependencies to work with node >= 18

### LINKED ISSUE #932: [CKEditor 5] import MathType >=8.8.1 leads to duplicated editors using CKEditor >=40.0.0 (version conflict)
## Description

When you import the MathType plugin in the CKEditor  `import MathType from '@wiris/mathtype-ckeditor5/src/plugin.js';`  in my `app.js` then make `npm update`, it imports a `@ckeditor` node module inside the `node_modules/@wiris` one, leading to a [ckeditor-duplicated-modules](https://ckeditor.com/docs/ckeditor5/latest/support/error-codes.html#error-ckeditor-duplicated-modules). 

## Environment

- CKEditor '>=40.0.0'.
- MathType '>= 8.8.1'.

Current version.

- CKEditor '41.2.1'.
-  MathType '8.8.3'
- Laravel '10.46' + PHP '8.1'.
-  NPM '9.2.0' + Vite.

## Steps to reproduce

Inside a laravel project, install the latest ckeditor **without** the online builder. 
(Following those steps [https://ckeditor.com/docs/ckeditor5/latest/installation/advanced/alternative-setups/integrating-from-source-vite.html#installing-necessary-dependencies](url) )
In `app.js` import the basic plugins to the ckeditor.

```
import ClassicEditor from '@ckeditor/ckeditor5-editor-classic/src/classiceditor';
...
import MathType from '@wiris/mathtype-ckeditor5/src/plugin.js';
```
Run `npm update` or `npm install` then `npm run build`.
Open the project in browser.

## Expected result

The CKEditor is shown with the MathType button.

## Actual result

Broken CKEditor. The javascript console triggered a ckeditor-duplicated-modules error.

## Other details

After investigation this issue is due to the version required in  `@wiris/mathtype-ckeditor5/package.json` 
```
...
"depedencies":{
    "@ckeditor/ckeditor5-core": "^38.0.0",
    "@ckeditor/ckeditor5-engine": "^38.0.0",
    "@ckeditor/ckeditor5-ui": "^38.0.0",
    "@ckeditor/ckeditor5-widget": "^38.0.0",
    ... 
}
```
But the latest ckeditor version is `>=40.0.0`, so npm makes a local copy of the ckeditor module inside the wiris node module when you do `npm install` because of the version conflict. 
MathType `8.8.0` works with CKEditor `>=40.0.0` since the requirement is `>= 23.0.0`.

### PR DESCRIPTION
**Pull request by carla-at-wiris**
## Description

Node `v17` introduced a breaking change. This means that, in order to use node ≥ `17.0.0`, all our devDependencies need to be compatible with the breaking change. We have some dependencies on the React and Angular demos which are not compatible with node ≥ 17.0.0. For that reason, we had to lock the CKEditor5 dependencies to version 38.0.0.

This PR includes the following:
* Updates all the possible project dependencies to their latest stable versions.
* Removes the creation of Angular & React staging demos because they are outdated and updating the dependencies is not enough, which also breaks the demos.
* Resolves the breaking changes for the updated dependencies, if any.
* Updates the staging glossary of the demos.

## Steps to reproduce

1. Go to the staging [demos glossary](https://integrations.wiris.kitchen/KB-45473/).
2. Validate that all the demos listed there are working properly.

---

Closes #932

[#taskid 45473](https://wiris.kanbanize.com/ctrl_board/2/cards/45473/details/)

### CODE REVIEW

**Comment by usantos-at-wiris** [comment]
Tested different actions with 
* HTML
  * CKEditor 4
  * CKEditor 5
  * Froala
  * Generic
  * TinyMCE 5
  * TinyMCE 6
* VUE 
  * CKEditor 5
 
All work correctly an github action pass all tests
image

Good job! 😊

**Review by usantos-at-wiris — APPROVED** [review]

**Comment by usantos-at-wiris** [comment]
Also tested running the development environments with node 21 and works perfect!

**Review by xjiang-at-wiris — APPROVED** [review]

**Review by jgonzalez-at-wiris — CHANGES_REQUESTED** [review]
Saquemos los cambios donde hemos comentado código a una tarea aparte donde hagamos limpieza de todo lo que no nos está aportando nada: cypress, el exclude de deploy-staging.yml, y el action que genera documentacion automáticamente si finalmente no somos capaces de hacerlo funcionar subiendo su versión de node.

File: .eslintrc.js
Comment (jgonzalez-at-wiris) [inline .eslintrc.js:145]: Este fichero comentado no se ha eliminado aunque dijimos de hacerlo, ¿es por algún motivo?

Comment (carla-at-wiris) [reply .eslintrc.js]: It was a mistake. Already removed!

**Review by jgonzalez-at-wiris — APPROVED** [review]

**Comment by even-steven** [comment]
Can someone please merge this if ready? I'm blocked on this issue.

**Comment by carla-at-wiris** [comment]
> Can someone please merge this if ready? I'm blocked on this issue.

Hello @even-steven, we're working on a release that will fix this issue and have it merged. We will let you know when it's merged and released.

If you need more assistance from us, I would suggest to please contact our Support Team at [support@wiris.com](mailto:support@wiris.com).

**Comment by carla-at-wiris** [comment]
@even-steven, the mentioned release is out with the changes of this pull request. The version is the `8.9.1`

**Comment by even-steven** [comment]
Thanks very much @carla-at-wiris and to those who made this fix 😄

**Pull request merged**
